localstack/utils/aws/aws_stack.py

```diff
@@ -20,6 +20,29 @@
 CACHE_S3_HOSTNAME_DNS_STATUS = None
 
 
+DEFAULT_PARTITION = "aws"
+REGION_PREFIX_TO_PARTITION = {
+    # (region prefix, aws partition)
+    "cn-": "aws-cn",
+    "us-gov-": "aws-us-gov",
+    "us-iso-": "aws-iso",
+    "us-isob-": "aws-iso-b",
+}
+PARTITION_NAMES = list(REGION_PREFIX_TO_PARTITION.values()) + [DEFAULT_PARTITION]
+ARN_PARTITION_REGEX = r"^arn:(" + "|".join(sorted(PARTITION_NAMES)) + ")"
+
+
+def get_partition(region: Optional[str]) -> str:
+    if not region:
+        return DEFAULT_PARTITION
+    if region in PARTITION_NAMES:
+        return region
+    for prefix in REGION_PREFIX_TO_PARTITION:
+        if region.startswith(prefix):
+            return REGION_PREFIX_TO_PARTITION[prefix]
+    return DEFAULT_PARTITION
```

Comment:
`aws_stack.py` is currently mostly legacy code, we try to replace everything in it step by step. I would like to avoid putting new code in here, but rather move it to either `arns.py` or perhaps some new module in the `util` package, `partition.py` for example.

Reply:
Done - all these methods are now in `arns.py`